nucypher/cli/commands/stake.py

```diff
@@ -1348,3 +1347,83 @@ def mint(general_config: GroupGeneralConfig,
                           emitter=emitter,
                           chain_name=blockchain.client.chain_name,
                           transaction_type='mint')
+
+    paint_staking_accounts(emitter=emitter,
+                           signer=STAKEHOLDER.signer,
+                           registry=STAKEHOLDER.registry,
+                           domain=STAKEHOLDER.domain)
+
+
+@stake.group()
+def rewards():
+    """Manage staking rewards."""
+
+
+@rewards.command()
+@group_staker_options
+@option_config_file
+@group_general_config
+@click.option('--period', help="Number of periods for which to calculate historical rewards", type=click.INT, default=0)
+def show(general_config, staker_options, config_file, period):
+    """Show staking rewards."""
+    if period < 0:
+        raise click.BadOptionUsage(option_name='--period', message='--period must positive')
+
+    emitter = setup_emitter(general_config)
+    STAKEHOLDER = staker_options.create_character(emitter, config_file)
+    _client_account, staking_address = select_client_account_for_staking(
+        emitter=emitter,
+        stakeholder=STAKEHOLDER,
+        staking_address=staker_options.staking_address)
+    blockchain = staker_options.get_blockchain()
+
+    staking_agent = STAKEHOLDER.staker.staking_agent
+
+    if period:
+        latest_block = blockchain.client.block_number
+        current_period = staking_agent.get_current_period()
+        seconds_per_period = staking_agent.staking_parameters()[0]
+        from_period = current_period - period
+        from_block = estimate_block_number_for_period(period=from_period,
+                                                      seconds_per_period=seconds_per_period,
+                                                      latest_block=latest_block)
+
+        argument_filters = {'staker': staking_address}
+        event_type = staking_agent.contract.events['Minted']
+        entries = event_type.getLogs(fromBlock=from_block, toBlock='latest', argument_filters=argument_filters)
+
+        reward_amount = 0
+        for event_record in entries:
+            reward_amount += event_record['args']['value']
+        reward_amount = NU(reward_amount, 'NuNit').to_tokens()
+    else:
+        reward_amount = STAKEHOLDER.staker.calculate_staking_reward().to_tokens()
+
+    emitter.echo(message=TOKEN_REWARD.format(reward_amount=reward_amount))
+
+
+@rewards.group()
+@group_staker_options
+@option_config_file
+@group_general_config
+@click.option('--tokens', help="Deposit NU tokens", type=click.STRING)
+@click.option('--amount', help="Amount to deposit", type=click.FLOAT)
+def deposit(general_config, staker_options, config_file, tokens, amount):
+    """Deposit rewards."""
```

Comment:
I cannot seem to find the original explainer comment - what will be the role of this command?

Reply:
There's an ongoing [discussion](https://github.com/nucypher/nucypher/issues/2509#issuecomment-817954554) about that, but I suspect it's superfluous.